lib/dry/types/constructor.rb

```diff
@@ -71,6 +71,13 @@ def constrained_type
         Constrained::Coercible
       end
 
+      # @api public
+      #
+      # @see Definition#to_ast
+      def to_ast
```

Comment:
for now just pass the fn as-is. @flash-gordon has an idea that we could register fn's explicitly in a registry, so that we can identify them easily. Having procs in AST is an anti-practice, but for now it's OK. We can improve it later once most of AST and a compiler are done.